src/parser.cpp

```diff
@@ -765,37 +767,56 @@ Writer<AST::AST*> Parser::parse_return_statement() {
 	return make_writer<AST::AST*>(e);
 }
 
-Writer<AST::AST*> Parser::parse_if_else_statement(AST::AST* parsed_condition) {
+Writer<AST::AST*> Parser::parse_if_else_stmt_or_expr() {
 	Writer<AST::AST*> result = {
 	    {"Parse Error: Failed to parse if-else statement"}};
 
-	Writer<AST::AST*> condition;
-	if (!parsed_condition) {
-		REQUIRE(result, TokenTag::KEYWORD_IF);
-		REQUIRE(result, TokenTag::PAREN_OPEN);
+	AST::AST* condition;
 
-		auto condition = parse_expression();
-		CHECK_AND_RETURN(result, condition);
+	REQUIRE(result, TokenTag::KEYWORD_IF);
+	if (consume(TokenTag::PAREN_OPEN)) {
+		auto expr = parse_expression();
+		CHECK_AND_RETURN(result, expr);
 		REQUIRE(result, TokenTag::PAREN_CLOSE);
+
+		auto paren = m_ast_allocator->make<AST::ParenthesizedExpression>();
+		paren->m_expr = expr.m_result;
```

Comment:
this looks exactly like inlining a piece of parse_terminal, why is calling parse_expression instead of writing this if-else here not good?

Reply:
I mean, (indirectly) almost the first thing parse_expression does is check for paren and start building a paren-expr

Reply:
also, does your code handle `if (func_name)(arg1, arg2) then value1 else value2` ?

Reply:
I just tested it, it does not.